daml-script/export/src/main/scala/com/daml/script/export/Main.scala

```diff
@@ -45,29 +45,35 @@ object Main {
       ec: ExecutionContext,
       esf: ExecutionSequencerFactory,
       mat: Materializer,
-  ): Future[Unit] =
-    for {
-      client <- LedgerClient.singleHost(config.ledgerHost, config.ledgerPort, clientConfig)
-      acs <- LedgerUtils.getACS(client, config.parties, config.start)
-      trees <- LedgerUtils.getTransactionTrees(client, config.parties, config.start, config.end)
-      acsPkgRefs = TreeUtils.contractsReferences(acs.values)
-      treePkgRefs = TreeUtils.treesReferences(trees)
-      pkgRefs = acsPkgRefs ++ treePkgRefs
-      pkgs <- Dependencies.fetchPackages(client, pkgRefs.toList)
-      _ = Dump.writeDump(
-        config.sdkVersion,
-        config.damlScriptLib,
-        config.outputPath,
-        acs,
-        trees,
-        pkgRefs,
-        pkgs,
-        config.acsBatchSize,
-      )
-    } yield ()
+  ): Future[Unit] = {
+    config.exportType match {
```

Comment:
Looks like a `fold`, but feel free to leave it as is if you think it's more readable.

Reply:
Yeah, personally I find the `match` version more readable. But, I have no strong opinion on this.